src/creation.cpp

```diff
@@ -0,0 +1,436 @@
+            loops.push_back(make_s2loop(ring, false));
+        }
+    }
+
+    auto polygon_ptr = std::make_unique<S2Polygon>();
+    polygon_ptr->set_s2debug_override(S2Debug::DISABLE);
+    polygon_ptr->InitNested(std::move(loops));
+
+    // Note: this also checks each loop of the polygon
+    if (!polygon_ptr->IsValid()) {
+        std::stringstream err;
+        S2Error s2err;
+        err << "polygon is not valid: ";
+        polygon_ptr->FindValidationError(&s2err);
+        err << s2err.text();
+        throw py::value_error(err.str());
+    }
+
+    return make_geography<s2geog::PolygonGeography>(std::move(polygon_ptr));
+}
+
+std::unique_ptr<Geography> geography_collection(const std::vector<Geography *> &features) {
+    std::vector<std::unique_ptr<s2geog::Geography>> features_copy;
+    features_copy.reserve(features.size());
+
+    for (const auto &feature_ptr : features) {
+        features_copy.push_back(feature_ptr->clone_geog());
+    }
+
+    return make_geography<s2geog::GeographyCollection>(std::move(features_copy));
+}
+
+//
+// ---- Geography creation Python bindings
+//
+
+void init_creation(py::module &m) {
+    py::options options;
+    options.disable_function_signatures();
+
+    m.def("point",
+          &point,
+          py::arg("longitude"),
+          py::arg("latitude"),
+          R"pbdoc(point(longitude: float, latitude: float) -> Geography
+
+        Create a POINT geography.
+
+        Parameters
+        ----------
+        longitude : float
+            longitude coordinate, in degrees.
+        latitude : float
+            latitude coordinate, in degrees.
+
+    )pbdoc");
+
+    m.def("points",
+          py::vectorize(&point),
+          py::arg("longitude"),
+          py::arg("latitude"),
+          R"pbdoc(
+        Create an array of points.
+
+        Parameters
+        ----------
+        longitude : array_like
+            longitude coordinate(s), in degrees.
+        latitude : array_like
+            latitude coordinate(s), in degrees.
+
+    )pbdoc");
+
+    m.def("points",
+          &points,
+          py::arg("coords"),
+          R"pbdoc(
+        Create an array of points.
+
+        Parameters
+        ----------
+        coords : array_like
+            A array of longitude, latitude coordinate tuples (i.e., with shape (N, 2)).
+
+    )pbdoc");
+
+    m.def("multipoint",
+          &multipoint<std::pair<double, double>>,
+          py::arg("points"),
+          R"pbdoc(multipoint(points: Sequence) -> Geography
+        Create a MULTIPOINT geography (overloaded function).
+
+        Create a MULTIPOINT geography from coordinates:
+
+        Parameters
+        ----------
+        points : sequence
+            A sequence of (longitude, latitude) coordinates, in degrees.
+
+    )pbdoc");
+
+    m.def("multipoint",
+          &multipoint<Geography *>,
+          py::arg("points"),
+          R"pbdoc(
+        Create a MULTIPOINT geography from POINT geographies:
```

Comment:
How does pybind11 handle this if you have an overloaded function (`multipoint` with list of coord pairs vs list of points), and you specify a docstring for both versions? 
For users ideally we have one docstring that mentions both options for the `points` argument?

Reply:
Pybind11 concatenates the docstrings and enumerate the signature of each overloaded function.

Since the arguments don't change but their types does, I disabled the auto-generated signatures and manually added a custom, generic one in the docstrings of the 1st overload (you can have a look at the rendered API documentation for this PR on RTD).

Alternatively we could define docstrings only for the 1st overload such that there is one "Parameters" section with mixed types.

Reply:
Seeing the example in https://spherely--51.org.readthedocs.build/en/51/_api_generated/spherely.multipoint.html, I think I would have a preference for manually writing a combined parameters section. But that is also something that can be addressed later!

Reply:
That wasn't much work to address so I did it here.